src/main/kotlin/org/rust/lang/core/resolve/RustResolveEngine.kt

```diff
@@ -298,10 +299,24 @@ private class Resolver {
 
         protected fun seek(elem: RustDeclaringElement) = seek(listOf(elem))
 
-        protected fun seek(decls: Collection<RustDeclaringElement>) {
-            decls.flatMap { it.boundElements }
-                .find { match(it) }
-                ?.let { found(it) }
+        protected fun seek(declaringElements: Collection<RustDeclaringElement>) {
+            for (element in declaringElements) {
+                check(matched == null)
+
+                // Recursively step into `use foo::*`
+                if (element is RustUseItem && element.isStarImport) {
+                    val pathPart = element.viewPath.pathPart ?: continue
+                    if (!addToSeen(pathPart)) {
+                        resolve(pathPart).element?.accept(this)
+                    }
+
+                    if (matched != null) {
+                        return
+                    }
+                } else {
+                    element.boundElements.find { match(it) }?.let { return found(it) }
```

Comment:
Ok, it's a little bit controversial: since return type is not specified (and we're assuming it as `Unit`) let's align this one with the whole thing: replacing `return found(it)` with just `found(it)`.